### ISSUE TITLE
openssh: dependency and linking against libfido2

### ISSUE THREAD

**Issue by mirko** [openwrt/packages#24509]
I'm having some questions re openssh's fido support.

tl;dr: why (only) the openssh-server depends on and links to libfido2 if fido support is selected + compiled in?

According to the package there's the config flag `OPENSSH_LIBFIDO2` which indicates whether openssh is build with `--with-security-key-builtin` or not.

As long as `--with-security-key-builtin` is passed, there's a dependency on `libfido2` - not just for the OpenWrt package, but the actual openssh build process requires `libfido2` once `--with-security-key-builtin` is set.

So far, so good.

However, within OpenWrt, the openssh-*server* is the only package depending libfido2.
Which appears weird, given that I'd rather have expected libfido2 being used on the *client*-side, e.g. for generation / transmission of fido2/webauth-keys (onto the HSM).

Also, looking at openssh-server packages from other distributions, none of those depend on libfido2, yet advertise for and work with fido keys (as in: fido-authentication against the openssh-server works). None of the sshd binaries is linked against libfido2 either.

Could someone shed some light onto this? Maybe its maintainer @tripolar ?
